client/core/trade.go

```diff
@@ -758,7 +789,7 @@ func (t *trackedTrade) swapMatches(matches []*matchTracker) error {
 // expires if the trade does not progress as expected.
 func (t *trackedTrade) finalizeSwapAction(match *matchTracker, coinID, contract []byte) error {
 	_, _, proof, auth := match.parts()
-	if auth.InitSig != nil {
+	if len(auth.InitSig) != 0 {
```

Comment:
Need similar check for `auth.RedeemSig` in `finalizeRedeemAction` - `auth.RedeemSig != nil` --> `len(auth.RedeemSig != nil) > 0`